ZukoAzula2016/src/org/team1540/zukoazula/Shooter.java

```diff
@@ -30,44 +32,101 @@
-        states.setStateWhen("backward", intakeArmRollerBackward.andNot(spinup));
-
-        states.getIsState("passive").toFloat(states.getIsState("forward").toFloat(-1, 1), 0).send(intakeArmRollerCAN.simpleControl());
-
-        FloatInput indexerIntakeSpeed = ZukoAzula.mainTuning.getFloat("Indexer Speed During Intake", 1f);
-        FloatInput indexerPassiveSpeed = ZukoAzula.mainTuning.getFloat("Roller Passive Speed", 0.1f);
-
-        rollerSpeed.attach(rollerArb.addBehavior("Not Shooting", FRC.inTeleopMode().and(shooter.isStopped)),
-                states.getIsState("passive").toFloat(indexerIntakeSpeed.negatedIf(states.getIsState("backward")), indexerPassiveSpeed));
-
-        BooleanInput shouldSpinUp = setupRollersForSpinup(spinup, actuallyFiring);
-
-        shooter.setup(shouldSpinUp);
-
-        fire.and(shooter.isUpToSpeed).onPress().send(actuallyFiring);
+        
+        StateMachine flywheelTarget = new StateMachine(0, "off", "low", "high");
+        StateMachine flywheelActual = new StateMachine(0, "off", "low", "high");
+        
+        StateMachine shooterStates = new StateMachine(0, 
+                "passive", // do nothing
+                "ejecting", // when the ball is being ejected from the intake mechanism
+                "intaking", // when the intake mechanism is intaking a ball
+                "loaded", // when a ball is loaded in the intake mechanism
+                "cocking", // when a ball is being cocked so that it is not in contact w/ the flywheel 
+                "spinup",  // when the flywheel is spinning up to maximum speed
+                "firing"); // when the ball is firing
+        
+        TalonExtendedMotor flywheel = makeLinkedTalons();
+        flywheel.modEncoder().configureEncoderCodesPerRev(125*15);
+        flywheel.modGeneralConfig().getBrakeNotCoast().set(false);
+        FloatInput flywheelSpeed = flywheel.modEncoder().getEncoderVelocity().absolute();
+        FloatOutput flywheelSimple = flywheel.simpleControl().addRamping(ZukoAzula.mainTuning.getFloat("Shooter Flywheel Ramping", 0.01f).get(), 
+                FRC.constantPeriodic);
+        
+        BooleanInput inhaleButton = ZukoAzula.controlBinding.addBoolean("Shooter Intake");
+        BooleanInput exhaleButton = ZukoAzula.controlBinding.addBoolean("Shooter Eject");
+        BooleanInput prefireButton = ZukoAzula.controlBinding.addBoolean("Shooter Spinup");
+        BooleanInput fireButton = ZukoAzula.controlBinding.addBoolean("Shooter Fire");
+        BooleanInput cancelShooterButton = ZukoAzula.controlBinding.addBoolean("Shooter Cancel Action");
+        
+        // Behavior
+        
+        flywheelSimple.setWhen(0.0f, FRC.duringTele.and(flywheelTarget.getIsState("off")));
+        flywheelSimple.setWhen(-0.068f, FRC.duringTele.and(flywheelTarget.getIsState("low")));
+        flywheelSimple.setWhen(1.0f, FRC.duringTele.and(flywheelTarget.getIsState("high")));
```

Comment:
This isn't a good way to do this. However, since there's currently no better way, I'm currently ramming a new method into devel-3.x.x on the CCRE, to arrive in a few minutes.